packages/zoe/src/contracts/attestation/attestation.js

```diff
@@ -0,0 +1,191 @@
+
+  const empty = AmountMath.makeEmpty(bldBrand, AssetKind.NAT);
+
+  const govAttManager = await setupExpiringAttestation(
+    'BLD-ATT-GOV',
+    empty,
+    zcf,
+  );
+  const locAttManager = await setupReturnableAttestation(
+    'BLD-ATT-LOC',
+    empty,
+    zcf,
+  );
+
+  /** @type {Max} */
+  const max = (x, y) => {
+    return AmountMath.isGTE(x, y) ? x : y;
+  };
+
+  /** @type {GetLienAmount} */
+  const getLienAmount = (address, currentTime) => {
+    const govLienAmount = govAttManager.getLienAmount(address, currentTime);
+    const locLienAmount = locAttManager.getLienAmount(address);
+    return max(govLienAmount, locLienAmount);
+  };
+
+  /**
+   * Get the amount currently liened per address. Must pass the
+   * current time.
+   *
+   * @param {Array<Address>} addresses
+   * @param {Timestamp} currentTime
+   * @returns {Array<Amount>}
+   */
+  const getLiened = (addresses, currentTime) => {
+    lastTime = currentTime;
+    return addresses.map(address => getLienAmount(address, currentTime));
+  };
+
+  const slash = (addresses, currentTime) => {
+    lastTime = currentTime;
+    return addresses.map(govAttManager.disallowExtensions);
+  };
+
+  const makeReturnAttInvitation = () =>
+    zcf.makeInvitation(locAttManager.returnAttestation, 'ReturnAtt', {
+      brand: locAttManager.brand,
+    });
+
+  const makeExtendAttInvitation = newExpiration => {
+    // Fail-fast if the newExpiration is already out of date compared
+    // to the last time we received from Cosmos
+    assert(
+      newExpiration > lastTime,
+      `The attestation could not be extended, as the new expiration ${newExpiration} is in the past`,
+    );
+    const offerHandler = seat =>
+      govAttManager.extendExpiration(seat, newExpiration);
+    return zcf.makeInvitation(offerHandler, 'ExtendAtt', {
+      brand: govAttManager.brand,
+    });
+  };
+
+  /** @type {MakeAttestationsInternal} */
+  const makeAttestationsInternal = async (
+    address,
+    amountToLien,
+    expiration,
+  ) => {
+    amountToLien = AmountMath.coerce(bldBrand, amountToLien);
+    assert.typeof(expiration, 'bigint');
+
+    /** @type {{total: Amount, bonded: Amount, locked: Amount,
+     * currentTime: Timestamp}} */
+    const accountState = await E(cosmos).getAccountState(address);
+    let { total, bonded, locked } = accountState;
+    const { currentTime } = accountState;
+    total = AmountMath.coerce(bldBrand, total);
+    bonded = AmountMath.coerce(bldBrand, bonded);
+    locked = AmountMath.coerce(bldBrand, locked);
+    assert.typeof(currentTime, 'bigint');
+    assert(
+      expiration > currentTime,
+      `Expiration ${expiration} must be after the current time ${currentTime}`,
+    );
+
+    const unlocked = AmountMath.subtract(total, locked);
+
+    // The amount for the address that currently has attestations that
+    // create a lien
+    const liened = getLienAmount(address, currentTime);
+    const unliened = AmountMath.subtract(total, liened);
```

Comment:
Good point. `AmountMath.subtract` errors if the left argument is less than the right. I'll change this to `subtractOrMakeEmpty` which subtracts only if the left argument is greater than or equal to the right, and otherwise returns the "empty" amount.